File: folia-api/src/main/java/net/azisaba/vanilife/event/DragonBuff.java
package net.azisaba.vanilife.event;

import org.bukkit.potion.PotionEffectType;
import org.jspecify.annotations.NullMarked;

@NullMarked
public final class DragonBuff {
    private final PotionEffectType type;
    private final int amplifier;
    private final int durationTicks;

    public DragonBuff(final PotionEffectType type, final int amplifier, final int durationTicks) {
        this.type = type;
        this.amplifier = amplifier;
        this.durationTicks = durationTicks;
    }

    public DragonBuff(final PotionEffectType type, final int amplifier) {
        this(type, amplifier, 200);
    }

    public PotionEffectType getType() {
        return this.type;
    }

    public int getAmplifier() {
        return this.amplifier;
    }

    public int getDurationTicks() {
        return this.durationTicks;
    }
}

File: plugins/plugin-islands/src/main/kotlin/net/azisaba/vanilife/islands/DragonBuff.kt
package net.azisaba.vanilife.islands

import org.bukkit.potion.PotionEffectType

/**
 * Simple DTO describing a potion-like buff provided by Dragon feature.
 */
data class DragonBuff(
    val type: PotionEffectType,
    val amplifier: Int = 0,
    val durationTicks: Int = 200 // default 10s
)
